src/api-binder/report.ts

```diff
@@ -128,6 +138,43 @@ async function reportCurrentState(opts: StateReportOpts, uuid: string) {
 	}
 }
 
+/**
+ * Cache last reported current state to CACHE_PATH
+ */
+async function cache(state: DeviceState) {
+	try {
+		await writeAndSyncFile(CACHE_PATH, JSON.stringify(state));
+	} catch (e: unknown) {
+		log.debug(`Failed to cache last reported state: ${(e as Error).message}`);
+	}
+}
+
+/**
+ * Get last cached state report from CACHE_PATH
+ */
+async function getCache(): Promise<DeviceState> {
+	try {
+		// Touch the file, which will create it if it doesn't exist
+		await touch(CACHE_PATH);
+
+		// Get last reported current state
+		const rawStateCache = await readFile(CACHE_PATH, 'utf-8');
+		const state = JSON.parse(rawStateCache);
+
+		// Return current state cache if valid
+		if (!DeviceState.is(state)) {
+			throw new Error();
```

Comment:
Shouldn't it delete the cache file if it's invalid rather than throwing and retrying forever?